### PR TITLE
Adding CERN's minuit solver

### LINKED ISSUE #166: Investigate work involved to add support for Minuit
This was suggested by Miguel Gonzales after presentation at SINE2020 workshop in Lund, May 2019.

This issue is to investigate the work involved in adding support for minimizers https://www.cern.ch/minuit at the level of how this tool can be tested with SciPy and Mantid minimizers. Discuss this outcome and if go ahead create the relevant issue(s) to do this

### PR DESCRIPTION
**Pull request by tyronerees**
#### Description of Work

Fixes #166 

#### Testing Instructions

1. Add `minuit` as a software in, e.g., `example_runScripts.py`
2. Check results are consistent with other minimizers

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: fitbenchmarking/fitting/controllers/minuit_controller.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        """        
+        # minuit requires an initial step size.
+        # The docs say
+        # "A good guess is a fraction of the initial 
+        #  fit parameter value, e.g. 10% 
+        #  (be careful when applying this rule-of-thumb
+        #  when the initial parameter value is zero "
+        self._initial_step = 0.1 * np.array(self.initial_params)
+        # set small steps to something sensible(?)
+        self._initial_step[self._initial_step < 1e-12] = 1e-12
```
Comment (wathen) [inline fitbenchmarking/fitting/controllers/minuit_controller.py]: Is `1e-12` what they suggest as a default tolerance too?

File: fitbenchmarking/fitting/controllers/minuit_controller.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            f = f / self.data_e
+
+        return np.dot(f,f)
+
+    def fit(self):
+        """
+        Run problem with Minuit
+        """
+        self.success = False
+        m = Minuit.from_array_func(self._prediction_error,
```
Comment (wathen) [inline fitbenchmarking/fitting/controllers/minuit_controller.py]: Does this make sense to put in the `setup` function so we just time `m.migrad()`?

File: fitbenchmarking/fitting/controllers/minuit_controller.py (hunk 1 above)
Comment (tyronerees) [reply fitbenchmarking/fitting/controllers/minuit_controller.py]: No -- it says:

> You can set initital step sizes with keywords, `error_<name> = <step size>`. If you don't know what to pick, don't worry too much. Usually, the choice is not important. In the worst case, iminuit will use a few more function evaluations to find the minimum. A good guess is a fraction of the initial fit parameter value, e.g. 10 % (be careful when applying this rule-of-thumb when your the initial parameter value is zero).

`1e-12` is plucked out of thin air, in the absence of any other information.

Comment (wathen) [reply fitbenchmarking/fitting/controllers/minuit_controller.py]: Maybe once the options file is completed we have a tolerance for things being close to zero

We kinda use it here as well
https://github.com/fitbenchmarking/fitbenchmarking/blob/1c6d59583ce7c7b75922dbed8235998bf4768c83/fitbenchmarking/fitting/controllers/base_controller.py#L74-L75

File: fitbenchmarking/fitting/controllers/minuit_controller.py (hunk 2 above)
Comment (tyronerees) [reply fitbenchmarking/fitting/controllers/minuit_controller.py]: Fixed.  I also moved the bit that saves as a scipy array into `cleanup`

**Pull request merged**